### PR TITLE
:recycle: Relocate client saving data in a file service

### PR DESCRIPTION
**Pull request by Bedhiver**
- Relocate client saving data in a file service
- Transform all static methods into instances

### CODE REVIEW

File: src/outputs/epic-games/client.ts [hunk 1]
```diff
@@ -1,38 +1,30 @@
...
-import fs from "fs";
 
-export default class ClientEpicGames {
-    private static clientInformations = clientInformations.epicGames;
-    private static clientInterface: ClientInterface = {
-        url: `${this.clientInformations.baseUrl}/${this.clientInformations.endpoint}`,
-        params: this.clientInformations.params
-    };
-    private static data: EpicGamesDatasInterface;
+export class ClientEpicGames {
```
Comment (anthonypillot) [inline src/outputs/epic-games/client.ts:6]: I think that it should be something like:
```suggestion
export class EpicGamesClient {
```
... if we want to follow the `SomethingInterface` or `GamesClient` rules.

File: src/outputs/epic-games/client.ts [hunk 2]
```diff
@@ -1,38 +1,30 @@
...
-                    console.log("Cache Updated !");
-                });
-                // TODO: Send mail with `elementsToSave`
-            }
-        } catch (err) {
-            console.log(err);
+            const data: EpicGamesDatasInterface = await axios.get(this.clientParamsConnection.url, this.clientParamsConnection.params);
+            this.clientService.updateCache(data);
+        } catch (error) {
+            console.log(error);
```
Comment (anthonypillot) [inline src/outputs/epic-games/client.ts:27]: Can we be more explicit? Better is the error wording, easily we'll retrieve it in the logs, no?
```suggestion
            console.log("An error has occurred while update EpicGames cache file", error);
```

File: src/services/client.service.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+     * Update `cache.json` with fresh datas from Epic Games
+     * @param {EpicGamesDatasInterface} data Datas from Epic Games
+     */
+    public async updateCache(data: EpicGamesDatasInterface): Promise<void> {
+        try {
+            const elementsToSave: ElementToSendInterface[] = EpicGamesMapperHelper.map(data);
+            
+            if (elementsToSave.length) {
+                fs.writeFile("src/data/cache.json", JSON.stringify(elementsToSave, null, 4), () => {
+                    console.log("Cache Updated !");
```
Comment (anthonypillot) [inline src/services/client.service.ts:17]: Same logic to be more explicit and relevant:
```suggestion
                    console.log("EpicGames cache file was updated");
```

**Review by anthonypillot — CHANGES_REQUESTED** [review]

File: src/services/client.service.ts (hunk 3 above)
Comment (Bedhiver) [reply src/services/client.service.ts]: Done

File: src/outputs/epic-games/client.ts (hunk 2 above)
Comment (Bedhiver) [reply src/outputs/epic-games/client.ts]: Done

File: src/outputs/epic-games/client.ts (hunk 1 above)
Comment (Bedhiver) [reply src/outputs/epic-games/client.ts]: Done

**Review by anthonypillot — APPROVED** [review]

**Pull request merged**